### ISSUE TITLE
Intercative variable replacment also supported for string commands

### ISSUE THREAD

**Issue by isidorn** [PowerShell/vscode-powershell#1315]
Hello friends, I am from the vscode team and some users are reporting an issue with debugging powershell. More datils can be found here https://github.com/Microsoft/vscode/issues/49390

After investigating I noticed that your command `SpecifyScriptArgs ` which is used to substitue values before debugging is returning an array of strings instead of a string. This was never officialy supported, and with the refactoring on the vscode side we are more strict and only allow strings.
Bottom line you should change a cast to array to simply return a string and not an array here https://github.com/PowerShell/vscode-powershell/blob/6c1e30e2a2f24731f394e55c364f5c09e11e7e6c/src/features/DebugSession.ts#L232

Make sure that all your commands which are used for launch.json configs are returning strings and not array.

**Comment by rkeithhill** [comment]
Thanks for the heads up. I'll make this change.

**Comment by isidorn** [comment]
Great, thanks

**Comment by eknraw** [comment]
@isidorn @rkeithhill Thank you.

**Comment by rkeithhill** [comment]
@isidorn @weinand Small problem.  We allow the user to delete all the text in the "specify args" dialog box:

![image](https://user-images.githubusercontent.com/5177512/39791348-63d3a1ec-52f8-11e8-9879-4140218585e8.png)

The problem is that if the user wipes out all the text (intending to pass no args), then the debugger never bothers to launch this config.  AFAICT, I have no control over this behavior.  By this point, our `resolveDebugConfiguration` has already been called and the debug adapter is never sent the launch request.

**Comment by isidorn** [comment]
@rkeithhill in that case your `SpecifyScriptArgs` command should return an empty string `''` and I believe everything should work fine. Did you try that out?
This will work since type of empty string is string and [this check](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/configurationResolver/node/variableResolver.ts#L105) should correctly pass.

**Comment by rkeithhill** [comment]
But if the input box is left blank, an empty string is what I see in the debugger and is what is returned to vscode??  I suspect that is why I had the `return new Array(text)` code there in the first place.

**Comment by isidorn** [comment]
@rkeithhill ok please double check and let us know what is unexpected exactly.

**Comment by avelis26** [comment]
Not trying to be that guy but, I downloaded and installed the latest "stable" build of Visual Studio Code, loaded the PowerShell extension, and tried to execute a PowerShell script with args and it's broken...

Not exactly a "stable" build so I'm wondering if any changes should be made to the deployment pipeline so obvious bugs like this don't find their way into "production". I'm happy to help as I've always wanted to assist in the open source world.

**Comment by weinand** [comment]
@avelis26 VS Code did not change any specified and documented functionality here. We just closed some undocumented loophole and broke the Powershell extension.
We are sorry about that but there are thousand of extension out there and we cannot test them all against using unspecfied functionality.

**Comment by rkeithhill** [comment]
@avelis26 Given that this is a UI feature (inputbox) that is used as part of starting up the debugger, it might be difficult to test.  But if you can find a way to automate such a test, that would be great!  Hopefully, we'll have this fixed and can get out a 1.7.1 maintenance release shortly.  We have a few other issues that cropped up in 1.7.0 that need to be fixed as well.

**Comment by avelis26** [comment]
I actually do a lot of automated UI testing and would love the challenge. I
don't suppose you have a hand list of use cases for me to get started on :)

Regards,
*Graham Pinkston*
M 425.320.1368 <+14253201368>
H 682.708.5430 <+16827085430>

**Comment by avelis26** [comment]
@weinand 
I understand not all extensions can be tested, but I feel Visual Studio Code is a Microsoft product and thus, major Microsoft extensions like PowerShell should probably have a few smoke tests prior to release.

I also understand that dream may never come true but like I said, I'm happy help in whatever way I can.

**Comment by weinand** [comment]
@avelis26 we expect extension authors to verify their extensions against the daily Insiders release (in fact that's exactly the main purpose of the Insiders release).
We cannot track what undocumented features extensions rely on deliberately or by accident.

**Comment by rkeithhill** [comment]
@isidorn @weinand OK we are returning an empty string - the empty string that inputbox returns to us.  The debug session is not initiated.  

![image](https://user-images.githubusercontent.com/5177512/39850066-b3642fea-53cc-11e8-956f-f53969617ea3.png)

It appears to me that VSCode is treating an empty string result the same as undefined.  In either case, the debug session is not initiated.

**Comment by rkeithhill** [comment]
This bug has been fixed but you will need VSCode 1.24 or an insiders daily build of 1.24 >= 5/11.

**Comment by tylerl0706** [comment]
Almost to a new release! Just a few internal processes that need to be taken care of this one time. Thank you for you patience 😊

**Comment by BlueArcherX** [comment]
Yes, please. This bug is killing us at work :)

**Comment by BlueArcherX** [comment]
@mpearon Some component of this fix was on VS Code, it was shipped in 1.24.  The remaining fix has to come with vscode-powershell 1.7.1.  I am not sure what the hold up is, it has been 2.5 months since 1.7.0, and very little movement is showing in the [milestones for 1.7.1](https://github.com/PowerShell/vscode-powershell/milestone/50)

**Comment by rkeithhill** [comment]
My understanding is the change to use named pipes instead of TCP ports, introduced a delay due to both effort to implement and subsequent internal review. There's also been ongoing work to improve the whole release process to make it simpler/quicker to roll releases in the future. 

If you were to take a look at merged PR's, you'd see there's been a lot of work done since 1.7.0.  I'm as anxious as everyone else to see an update released but that has to be done by @tylerl0706 (or @rjmholt).  @tylerl0706 is out this week on a cruise (lucky dog).

**Comment by tylerl0706** [comment]
The PowerShell extension has been updated (now at 1.8.1). Can you give this a try now?

**Comment by jasoncopseynielsen** [comment]
I received notice of an update to the extension and after installation - I can confirm that I am once again able to run powershell with args in debugging. 

Thanks for sorting this out.

**Comment by 123dev** [comment]
Working fine in 1.8.1.

Thanks

**Comment by mpearon** [comment]
I'm still having issues with this - but only when the vsLiveShare extension is enabled.  I mentioned this thread when reporting the issue there.
